packages/ui/origin-ui-core/src/components/Icons/DeviceIcon.tsx

```diff
@@ -1,40 +1,44 @@
 import React from 'react';
 import { useTranslation } from 'react-i18next';
-import solar from '../../../assets/icon_solar.svg';
-import wind from '../../../assets/icon_wind.svg';
-import hydro from '../../../assets/icon_hydro.svg';
-import iconThermal from '../../../assets/icon_thermal.svg';
-import iconSolid from '../../../assets/icon_solid.svg';
-import iconLiquid from '../../../assets/icon_liquid.svg';
-import iconGaseous from '../../../assets/icon_gaseous.svg';
-import iconMarine from '../../../assets/icon_marine.svg';
+import { ReactComponent as IconWind } from '../../../assets/device/icon-wind.svg';
+import { ReactComponent as IconHydro } from '../../../assets/device/icon-hydro.svg';
+import { ReactComponent as IconThermal } from '../../../assets/device/icon-thermal.svg';
+import { ReactComponent as IconSolid } from '../../../assets/device/icon-solid.svg';
+import { ReactComponent as IconLiquid } from '../../../assets/device/icon-liquid.svg';
+import { ReactComponent as IconGaseous } from '../../../assets/device/icon-gaseous.svg';
+import { ReactComponent as IconMarine } from '../../../assets/device/icon-marine.svg';
+import { ReactComponent as IconSolar } from '../../../assets/device/icon-solar.svg';
 
 export function DeviceIcon(props: { type: string; className: string }): JSX.Element {
     const { t } = useTranslation();
     const { type, className } = props;
-    let image = solar;
+    let SvgImage = <IconSolar />;
     let tooltip = '';
 
     if (type.startsWith('Wind')) {
```

Comment:
You can move this condition into a function, and call it inside the component return statement removing variables assigned with `let`. 
You will have 2 separate functions: 1 for assigning tooltip and 1 for choosing matching Icon.